lib/spack/spack/compilers/__init__.py

```diff
@@ -154,6 +154,14 @@ def add_compilers_to_config(compilers, scope=None, init_config=True):
     """
     compiler_config = get_compiler_config(scope, init_config)
     for compiler in compilers:
+        if not compiler.cc:
+            tty.warn(f"{compiler.spec} does not have a C compiler")
+        if not compiler.cxx:
+            tty.warn(f"{compiler.spec} does not have a C++ compiler")
+        if not compiler.f77:
+            tty.warn(f"{compiler.spec} does not have a Fortran77 compiler")
+        if not compiler.fc:
+            tty.warn(f"{compiler.spec} does not have a Fortran compiler")
```

Comment:
I think this is going to get very verbose for macos users, since they generally have an `apple-clang` compiler configured with no fortran.

Reply:
The warning only comes when you add to your compilers.yaml, which is not too often. I could only warn about a missing C compiler or only about `%intel` compiler. Would any of those be ok?